metals/src/main/scala/scala/meta/internal/metals/StatusBar.scala

```diff
@@ -36,7 +36,7 @@ import org.eclipse.lsp4j.MessageType
  *   as long as the attached `Future[_]` is not completed.
  */
 final class StatusBar(
-    client: () => MetalsLanguageClient,
```

Comment:
Just to be sure, we did this before because we wanted to ensure we got the configured client, but in this scenario, we _always_ have the configured client right?

Reply:
Exactly. Previously, language client was obtained and set after creating MetalsLanguageServer - https://github.com/scalameta/metals/blob/3cbb04ef03d2d390b228744aa5c5824aa6a8274e/metals/src/main/scala/scala/meta/metals/Main.scala#L68-L70.

Now, with MetalsLanguageServer being refactored and new MetalsLspService, the later one is created with workspace and proper client passed in so there is no need for `() => ` workarounds

Reply:
[Here](https://github.com/kpodsiad/metals/blob/908c0c1e199626abcf3375d96555772568e794ab/metals/src/main/scala/scala/meta/metals/MetalsLanguageServer.scala#L176-L192), we're creating MetalsService